crates/re_types_builder/src/objects.rs

```diff
@@ -51,19 +51,50 @@ impl Objects {
         }
 
         // resolve objects
-        for obj in schema
-            .objects()
-            .iter()
-            // NOTE: Wrapped scalar types used by unions, not actual objects: ignore.
-            .filter(|obj| !obj.name().starts_with("fbs.scalars."))
-        {
+        for obj in schema.objects().iter() {
             let resolved_obj = Object::from_raw_object(include_dir_path, &enums, &objs, &obj);
             resolved_objs.insert(resolved_obj.fqname.clone(), resolved_obj);
         }
 
-        Self {
+        let mut this = Self {
             objects: resolved_enums.into_iter().chain(resolved_objs).collect(),
+        };
+
+        // Resolve field-level semantic transparency recursively.
+        let mut done = false;
+        while !done {
+            done = true;
+            let mut objects_copy = this.objects.clone(); // borrowck, the lazy way
+            for obj in this.objects.values_mut() {
+                for field in &mut obj.fields {
+                    if field.is_transparent() {
+                        if let Some(target_fqname) = field.typ.fqname() {
+                            let mut target_obj = objects_copy.remove(target_fqname).unwrap();
+                            assert!(
+                                target_obj.fields.len() == 1,
+                                "field '{}' is marked transparent but points to object '{}' which \
+                                    doesn't have exactly one field (found {} fields instead)",
+                                field.fqname,
+                                target_obj.fqname,
+                                target_obj.fields.len(),
+                            );
+
+                            field.typ = target_obj.fields.pop().unwrap().typ;
```

Comment:
need to override the field's attributes too, otherwise this will always recurse all the way to the bottom!